liblouis/compileTranslationTable.c

```diff
@@ -800,248 +790,235 @@ passFindCharacters(const FileInfo *file, widechar *instructions, int end,
 		// this character to it (possibly overwriting previous definition rules)
 		// adding the attributes to the character has already been done elsewhere
-		if (newRule->opcode >= CTO_Space && newRule->opcode < CTO_UpLow)
-			character->definitionRule = newRuleOffset;
+		if (rule->opcode >= CTO_Space && rule->opcode < CTO_UpLow)
+			character->definitionRule = ruleOffset;
 	}
 	// add the new rule to the list of rules associated with this character
 	// if the new rule is a character definition rule, it is inserted at the end of the
 	// list
 	// otherwise it is inserted before the first character definition rule
-	currentOffsetPtr = &character->otherRules;
-	while (*currentOffsetPtr) {
-		currentRule = (TranslationTableRule *)&(*table)->ruleArea[*currentOffsetPtr];
-		if (currentRule->charslen == 0) break;
-		if (currentRule->opcode >= CTO_Space && currentRule->opcode < CTO_UpLow)
-			if (!(newRule->opcode >= CTO_Space && newRule->opcode < CTO_UpLow)) break;
-		currentOffsetPtr = &currentRule->charsnext;
+	TranslationTableOffset *otherRule = &character->otherRules;
+	while (*otherRule) {
+		TranslationTableRule *r = (TranslationTableRule *)&(*table)->ruleArea[*otherRule];
+		if (r->charslen == 0) break;
+		if (r->opcode >= CTO_Space && r->opcode < CTO_UpLow)
+			if (!(rule->opcode >= CTO_Space && rule->opcode < CTO_UpLow)) break;
+		otherRule = &r->charsnext;
 	}
-	newRule->charsnext = *currentOffsetPtr;
-	*currentOffsetPtr = newRuleOffset;
+	rule->charsnext = *otherRule;
+	*otherRule = ruleOffset;
 }
 
 static void
-addForwardRuleWithMultipleChars(TranslationTableOffset newRuleOffset,
-		TranslationTableRule *newRule, TranslationTableHeader *table) {
-	/* direction = 0 newRule->charslen > 1 */
-	TranslationTableRule *currentRule = NULL;
-	TranslationTableOffset *currentOffsetPtr =
-			&table->forRules[_lou_stringHash(&newRule->charsdots[0], 0, NULL)];
-	while (*currentOffsetPtr) {
-		currentRule = (TranslationTableRule *)&table->ruleArea[*currentOffsetPtr];
-		if (newRule->charslen > currentRule->charslen) break;
-		if (newRule->charslen == currentRule->charslen)
-			if ((currentRule->opcode == CTO_Always) && (newRule->opcode != CTO_Always))
-				break;
-		currentOffsetPtr = &currentRule->charsnext;
+addForwardRuleWithMultipleChars(TranslationTableOffset ruleOffset,
+		TranslationTableRule *rule, TranslationTableHeader *table) {
+	/* direction = 0 rule->charslen > 1 */
+	TranslationTableOffset *forRule =
+			&table->forRules[_lou_stringHash(&rule->charsdots[0], 0, NULL)];
+	while (*forRule) {
+		TranslationTableRule *r = (TranslationTableRule *)&table->ruleArea[*forRule];
+		if (rule->charslen > r->charslen) break;
+		if (rule->charslen == r->charslen)
+			if ((r->opcode == CTO_Always) && (rule->opcode != CTO_Always)) break;
+		forRule = &r->charsnext;
 	}
-	newRule->charsnext = *currentOffsetPtr;
-	*currentOffsetPtr = newRuleOffset;
+	rule->charsnext = *forRule;
+	*forRule = ruleOffset;
```

Comment:
This seems quite similar to the code above for a single char. There you name the variable `otherRule`. Why do you use `forRule` here?

Reply:
I call it `forRule` because it comes from `table->forRules`. These are the forward rules that match multiple characters, they are a direct field of the table. 

What I call `otherRule` in the other place comes from `character->otherRules` (or `dots->otherRules`). These are rules that match a single character or dot pattern, they are a field of a character object.

So there is a real difference.

Reply:
OK, didn't know that. Makes sense then.